lib/metricAggregators.js

```diff
@@ -29,12 +79,20 @@ function AggregatorFactory(aggregatorFn) {
 		byLabels.forEach(values => {
 			if (values.length === 0) return;
 			const valObj = {
-				value: aggregatorFn(values),
 				labels: values[0].labels,
 			};
 			if (values[0].metricName) {
 				valObj.metricName = values[0].metricName;
 			}
+			// Should always `sum` if it's `sum` or `count` that's part of histogram & summary
+			if (
+				values[0].metricName === `${result.name}_sum` ||
+				values[0].metricName === `${result.name}_count`
+			) {
+				valObj.value = sum(values);
+			} else {
+				valObj.value = aggregatorFn(values);
+			}
```

Comment:
This is a good find, thanks and sorry for the year-long delay reviewing it.

I think this can wrongly identify custom metrics that are happen to be named `something_sum`/`something_count` though. Can you use the `type` instead? (`metrics[0].type`)